Fix source groups

arnaldo2792 opened the pull request:
**Description of changes:**

`early-boot-config` was missing `api` and `imdsclient` from its `source-group` dependencies, so changes to either `apiclient` or `imdsclient` didn't trigger a rebuild of `early-boot-config`.

`pluto` 's `Cargo.toml` included `source-groups` by mistake, and it doesn't have any effect.

**Testing done:**

After this change, I confirmed that `early-boot-config` was re-compiled whenever I changed `apiclient` or `imdsclient` (even by just `touch`ing the files). Before this change, `touch`ing didn't trigger a rebuild.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

arnaldo2792 committed:
early-boot-config: add missing source-groups dependencies

Add missing dependencies for early-boot-config so that it is rebuild
whenever they change

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
pluto: remove unnecessary source-groups from sources' Cargo.toml

`source-groups` should only be used within `packages/**/Cargo.toml`, in
any other place it doesn't have any effect

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

bcressey commented on packages/early-boot-config/Cargo.toml:
```diff
@@ -10,6 +10,8 @@ path = "../packages.rs"
 [package.metadata.build-package]
 source-groups = [
+    "api",
+    "imdsclient",
```
Let's just list `early-boot-config` here instead of the individual subdirectories.

Appears to be missing `constants`, `generate-readme`, `imdsclient`,  ...

The brute force way to generate this list would be:
* copy the sources tree to another location
* delete everything from the sources tree that isn't a directory in `source-groups`
* trim the workspace `Cargo.toml` and regenerate `Cargo.lock`
* see if the package still builds

`static-pods` and `netdog` probably also need to be revisited with an eye to completing the dependency picture.

arnaldo2792 committed:
netdog: add missing source-groups dependencies

Add missing dependencies for netdog so that it is rebuild whenever they
change

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
os: add missing source-groups dependencies

Add missing dependencies for the os package so that it is rebuild whenever
they change

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
static-pods: add missing source-groups dependencies

Add missing dependencies for static-pods so that it is rebuild whenever
they change

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 commented:
Forced push includes:
- Changes in `os/Cargo.toml` to add missing dependencies for rebuild
- Changes in `netdog/Cargo.toml` to add missing dependencies for rebuild
- Changes in `static-pods/Cargo.toml` to add missing dependencies for rebuild
